shared/metrics/src/test/java/io/pravega/shared/metrics/MetricRegistryUtils.java

```diff
@@ -9,38 +9,30 @@
  */
 package io.pravega.shared.metrics;
 
-import com.codahale.metrics.Counter;
-import com.codahale.metrics.Gauge;
-import com.codahale.metrics.Meter;
-import com.codahale.metrics.Metric;
-import com.codahale.metrics.Timer;
+import io.micrometer.core.instrument.Counter;
+import io.micrometer.core.instrument.Gauge;
+import io.micrometer.core.instrument.Metrics;
+import io.micrometer.core.instrument.Timer;
+import io.micrometer.core.instrument.DistributionSummary;
+
 import lombok.extern.slf4j.Slf4j;
 
 @Slf4j
 public class MetricRegistryUtils {
 
-    public static Counter getCounter(String metricsName) {
-        return (Counter) getMetric(metricsName);
-    }
-
-    public static Meter getMeter(String metricsName) {
-        return (Meter) getMetric(metricsName);
+    public static Counter getCounter(String metricsName, String... tags) {
+        return Metrics.globalRegistry.find(metricsName).tags(tags).counter();
     }
 
-    public static Gauge getGauge(String metricsName) {
-        return (Gauge) getMetric(metricsName);
+    public static DistributionSummary getMeter(String metricsName, String... tags) {
+        return Metrics.globalRegistry.find(metricsName).tags(tags).summary();
```

Comment:
Any chance some of this calls will return `null` along the way?

Reply:
Same everywhere below.

Reply:
No, this is of builder pattern, along the way it always return builder object (in this case Search).
Finally it may return null metric object, which is expected.